app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt

```diff
@@ -225,19 +241,24 @@ private fun assertSearchEngineURL(searchEngineName: String) {
         .check(matches(ViewMatchers.withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
 }
 
-private fun assertSearchEngineResults(searchEngineName: String) {
-    val count =
-        mDevice.wait(Until.findObjects(By.text((searchEngineName))), TestAssetHelper.waitingTime)
-    assert(count.size > 1)
+private fun assertSearchEngineResults(rule: ComposeTestRule, searchEngineName: String, count: Int) {
+    rule.onAllNodesWithText(searchEngineName)
+        .assertCountEquals(count)
 }
 
-private fun assertSearchView() {
-    onView(withId(R.id.search_wrapper)).check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
-}
+private fun assertSearchView() =
+    assertTrue(
+        mDevice.findObject(
+            UiSelector().resourceId("$packageName:id/search_wrapper")
+        ).waitForExists(waitingTime)
+    )
 
 private fun assertBrowserToolbarEditView() =
-    onView(Matchers.allOf(withId(R.id.mozac_browser_toolbar_edit_url_view)))
-        .check(matches(ViewMatchers.withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
+    assertTrue(
+        mDevice.findObject(
+            UiSelector().resourceId("$packageName:id/mozac_browser_toolbar_edit_url_view")
+        ).waitForExists(waitingTime)
+    )
 
 private fun assertScanButton() =
     onView(allOf(withText("Scan")))
```

Comment:
I think we need to change that to use UIAutomator as the function above this one

Reply:
And same for `assertSearchEngineButton` below